setup.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+usage="Usage: ./setup.sh </path/to/venv/>"
+
+# Check if path for venv folder provided
+if [[ -z "$1" ]]; then
+    echo -e "Missing required argument.\n$usage\n"
+    exit 1
+fi
+venv_path=$1
+
+# Check if venv is installed, install if not
+if [[ -z $(apt list --installed 2>/dev/null | grep python3-venv) ]]; then
+    echo -e "Missing required package python3-venv. Please run \"sudo apt update && sudo apt install python3-venv\""
+    exit 1
+fi
+
+# Create virtual environment at path
+if [[ -z $(ls $venv_path 2>/dev/null)]]; then
+    python3 -m venv $venv_path
```

Comment:
I think we could/should remove this check and run `python3 -m venv $venv_path` unconditionally. Our current check doesn't work if the desired path is an existing directory with contents other than the virtual environment. According to https://docs.python.org/3/library/venv.html#creating-virtual-environments, it will just reuse the existing venv if one was already created at the specified location.

If we really wanted to check that there isn't already a venv at the location to save a few seconds, we could switch the condition to `-z $(ls $venv_path 2>/dev/null | grep "pyvenv\.cfg")`

Reply:
Agreed